taichi/analysis/gather_uniquely_accessed_pointers.cpp

```diff
@@ -34,6 +34,11 @@ class LoopUniqueStmtSearcher : public BasicStmtVisitor {
       loop_unique_[stmt] = stmt->index;
   }
 
+  void visit(GlobalThreadIndexStmt *stmt) override {
```

Comment:
A thread index is not unique among the top-level loop. Consider the case that one sets the number of cpu threads to 1. So IMHO we shouldn't do this here, and we don't need to record the loop in this statement.

Reply:
I pressumed marking it as unique in the top level might mean that the statement is optimized in the lower-level loops but preserved in the top-level?

Reply:
`loop_unique` was introduced in #1961 and the intention was to mark/analyze those stmts with definitely different values across loop iterations so that no atomic operations need to involve when those stmts serve as field indices.